_rules/zoom-text-no-overflow-clipping-59br37.md

```diff
@@ -0,0 +1,247 @@
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Wilco Fiers
+  citations:
+    - The Raven, poem by Edgar Allan Poe
+---
+
+## Applicability
+
+The rule applies to any [text node][] for which all of the following is true when in a [viewport size][] of 640 by 512:
+
+- The [text node][] is [visible][]; and
+- The [text node][] has an HTML element as a [parent][] in the [flat tree][]; and
+- The [text node][] has an [ancestor][] in the [flat tree][] with the a [computed][] [overflow-x][overflow] or [overflow-y][overflow] of `hidden` or `clip`; and
+- The [text node][] does not have an [ancestor][] in the [flat tree][] with an `aria-hidden` attribute set to `true`
+
+**Note**: A [viewport size][] of 640 by 512 is equivalent to a [viewport size][] of 1280 by 1024 zoomed 200%.
+
+## Expectation
+
+Each test target is not [clipped by overflow][clipped] of an [ancestor][] in the [flat tree][] when in a [viewport size][] of 640 by 512, except if the [clipping][clipped] [ancestor][] has one of the following:
+
+- **text-overflow**: A [computed][] [white-space][] of `nowrap`, and a [computed][] [text-overflow][] that is not `clip`; or
+
+- **line wrapping**: A [computed][] [line-height][] equal to or greater than its
+  height of its [bounding box][], or in case of a [computed][] [overflow][] of `clip`, its [content box][].
+
+## Assumptions
+
+If any of the following assumptions is true, failing this rule may not result in a failure of [success criterion 1.4.4 Resize text](https://www.w3.org/TR/WCAG21/#resize-text):
+
+- There is no other mechanism for resizing text available on the page, that can be used to resize text to 200% without loss of information or functionality. This includes font resizing in the browser, or a javascript mechanism of resizing in the page.
+
+- [Text nodes][text node] can not be [clipped by overflow][clipped] without loss of information, except for [text nodes][text node] with an [ancestor][] with `aria-hidden` set to `true`, or when specific styles have been applied to ensure text is clipped cleanly (text-overflow, line wrapping or hidden text).
+
+- While [success criterion 1.4.4 Resize text](https://www.w3.org/TR/WCAG21/#resize-text) does not explicitly mention which viewport size has to be resized up to 200%, it is assumed that a [viewport size][] of 1280 by 1024 is applicable. A 1280 by 1024 [viewport size][] is explicitly mentioned under [success criterion 1.4.10 Reflow](https://www.w3.org/TR/WCAG21/#reflow).
+
+## Accessibility Support
+
+_No accessibility support issues known._
+
+## Background
+
+- [Understanding Success Criterion 1.4.4: Resize text](https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html)
+- [G42: Using a technology that has commonly-available user agents that support zoom](https://www.w3.org/WAI/WCAG21/Techniques/general/G142)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This [text node][] is fully [visible][] at a [viewport size][] of 640 by 512.
```

Comment:
Can you clarify what you mean? The browser's default stylesheet is included in the CSS spec. Obviously if a browser doesn't follow the CSS specs, its results will be different, but all browsers I know of use this default; at least where font-size is concerned they do.

Reply:
The base font size is not included in the suggested user agent styles of the HTML specification, no: https://html.spec.whatwg.org/multipage/rendering.html. It's true that modern browsers default to a base font size of 16 pixels, but this is entirely configurable:

![image](https://user-images.githubusercontent.com/1174718/80476578-ddcce780-894a-11ea-9b9d-742172353433.png)

Above is a screenshot of the font settings in Chrome.

Reply:
As @kasperisager points out, the spec does not specify the base font size. Any user can configure it. When I set the base font size to 24 pixels, the text in the first passed example was clipped (so, the example should fail in that instance). This can be easily solved by setting the font-size in the div's style. 

Not relying on the default styles is similar to what we have in other rules (e.g. resetting the style of every link to make sure that default styles where being used for links when looking at distinguishing styles). 

I would be happy with a solution that sets the font-size in this example (this issue does not impact the other examples).

Reply:
I don't think it's necessary, but I don't mind having font-size added to the examples, so I've done that.